src/keyboardNavigation.ts

```diff
@@ -0,0 +1,83 @@
+import { isDeepestVisible, isImmediateParent } from './utils';
+
+function findDeepestVisibleChild(node: HTMLElement): HTMLElement {
+    if (isDeepestVisible(node)) return node;
+    const ul = <HTMLElement>node.children[1];
+    return findDeepestVisibleChild(<HTMLElement>ul.children[ul.children.length - 1]);
+}
+
+function findNearestParentWithAdditionalLeaves(node: HTMLElement): HTMLElement {
+    const ul = node.parentElement;
+    if (!ul) return node;
+    const parent = ul.parentElement;
+    if (!parent || !parent.parentElement) return node;
+    const parentAndSiblings = [...parent.parentElement.children];
+    if (parentAndSiblings.indexOf(parent) == parentAndSiblings.length - 1) {
+        return findNearestParentWithAdditionalLeaves(parent);
+    } else {
+        return <HTMLElement>parentAndSiblings[parentAndSiblings.indexOf(parent) + 1];
+    }
+}
+
+export const navigateUp = (path: string, shadowRoot: ShadowRoot): void => {
+    try {
```

Comment:
i think when we get to making a PR, we probably shouldn't rely on try/catches there. in general your code is cautious about triggering runtime errors so i think maybe we just need to make one quick review at the end to be sure

Reply:
:sweat_smile:  can you tell I'm worried about runtime errors. I'm ok to remove them at the end if/when we are sure, but I'm leaving for now

Reply:
removed them :fearful: